### PR TITLE
Add charmbrowser

### PR DESCRIPTION
**Pull request by huwshimi**
Add a charmbrowser component to display search results/store/mid-point views depending on the app state.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- view `http://0.0.0.0:6543/:flags:/react/`
- everything should appear as normal
- do a search from the header
- the search results should appear
- add `&midpoint` to the url (e.g. `http://0.0.0.0:6543/:flags:/react/?text=django&midpoint`)
- you should see the mid-point

There will be followup work to correctly set the sectionC state that will make these components appear correctly.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3064/
Test PASSed.

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('charmbrowser-component', function() {
+
+  juju.components.Charmbrowser = React.createClass({
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: Nice - once we flesh this out a little more I wonder if there is a mixin we can create to share some logic between the charmbrowser and inspector components.

**Comment by hatched** [comment]
:+1: QA :ok: ! This looks great!

**Comment by jaycee** [comment]
@huwshimi Looks good, thanks! :+1:

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1056

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**